File: frontend/src/types/portfolio.ts
export interface PortfolioSettings {
  bio: string | null
  portfolio_public: boolean
}

export interface PortfolioProject {
  project_title: string
  project_slug: string
  submission_url: string | null
  reviewed_at: string | null
}

export interface PortfolioCertificate {
  course_title: string
  certificate_number: string
  issued_at: string
}

export interface PublicPortfolio {
  user_id: string
  display_name: string
  bio: string | null
  projects: PortfolioProject[]
  certificates: PortfolioCertificate[]
}

File: backend/app/schemas/portfolio.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PortfolioSettings(BaseModel):
    bio: str | None
    portfolio_public: bool


class PortfolioSettingsUpdate(BaseModel):
    bio: str | None = None
    portfolio_public: bool | None = None


class PortfolioProject(BaseModel):
    project_title: str
    project_slug: str
    submission_url: str | None
    reviewed_at: datetime | None


class PortfolioCertificate(BaseModel):
    course_title: str
    certificate_number: str
    issued_at: datetime


class PublicPortfolio(BaseModel):
    user_id: uuid.UUID
    display_name: str
    bio: str | None
    projects: list[PortfolioProject]
    certificates: list[PortfolioCertificate]
